library/src/main/java/com/novoda/downloadmanager/DownloadBatchRequirementRules.java

```diff
@@ -21,12 +23,27 @@ void addRule(DownloadBatchRequirementRule storageRequirementRule) {
 
     @Override
     public boolean hasViolatedRule(DownloadBatchStatus downloadBatchStatus) {
+        return getViolatedRule(downloadBatchStatus).isPresent();
+    }
+
+    @Override
+    public Optional<DownloadBatchRequirementRule> getViolatedRule(DownloadBatchStatus downloadBatchStatus) {
         for (DownloadBatchRequirementRule requirementRule : rules) {
             if (requirementRule.hasViolatedRule(downloadBatchStatus)) {
-                return true;
+                return Optional.of(requirementRule);
             }
         }
 
-        return false;
+        return Optional.absent();
+    }
+
+    /**
+     * @return [null] - a composite rule isn't identifiable by itself.
+     * Retrieve the code of the actually violated rule via {@link #getViolatedRule#getCode()}.
```

Comment:
javadoc should be with the interface, not the implementation

Reply:
The interface where this method is implemented is a *grandparent* of this class. From this grandparent's perspective there is no notion of a "composite" rule, so this comment would be out of place there. 

It's only the "plural" version that should return null for this value - and not every ``DownloadBatchRequirementRule` will be plural.